task-manager-backend/src/middlewares/error-middleware.js

```diff
@@ -0,0 +1,18 @@
+const ApiError = require("../utils/api-error");
+
+const notFound = (req, res, next) => {
+    next(new ApiError(404, `404 Not Found!! ${req.originalUrl}`));
+};
+
+const errorHandler = (err, req, res, next) => {
+    const { statusCode, message, success, errors } = err;
```

Comment:
why do we need this?
Can we do like:
res.status(err.statusCode || 500).json({
        statusCode: err.statusCode || 500,
        message: err.message,
        success: err.success,
        errors: err.errors,
    });

Reply:
I have just added because statusCode is used two places and that 'or' logic is also needed.

I will change this.